java/code/src/com/redhat/rhn/frontend/xmlrpc/saltkey/SaltKeyHandler.java

```diff
@@ -37,25 +40,120 @@ public SaltKeyHandler(SaltKeyUtils saltKeyUtilsIn) {
         this.saltKeyUtils = saltKeyUtilsIn;
     }
 
+    /**
+     * API endpoint to list accepted salt keys
+     * @param loggedInUser the user
+     * @return 1 on success
+     *
+     * @xmlrpc.doc List accepted salt keys
+     * @xmlrpc.param #param("string", "loggedInUser")
+     * @xmlrpc.param #param("string", "minionId")
+     * @xmlrpc.returntype #array_single("string", "Accepted salt key list")
+     */
+    public List<String> acceptedList(User loggedInUser) {
+        try {
+            ensureOrgAdmin(loggedInUser);
+        }
+        catch (PermissionException e) {
+            throw new PermissionCheckFailureException(e);
+        }
+        return saltKeyUtils.acceptedSaltKeyList(loggedInUser);
+
+    }
+
+
+    /**
+     * API endpoint to list pending salt keys
+     * @param loggedInUser the user
+     * @return 1 on success
+     *
+     * @xmlrpc.doc List pending salt keys
+     * @xmlrpc.param #param("string", "loggedInUser")
+     * @xmlrpc.returntype #array_single("string", "Pending salt key list")
+     */
+    public List<String> pendingList(User loggedInUser) {
+        try {
+            ensureOrgAdmin(loggedInUser);
+        }
+        catch (PermissionException e) {
+            throw new PermissionCheckFailureException(e);
+        }
```

Comment:
I think a try/catch is not needed here as ensureOrgAdmin() throw already a PermissionCheckFailureException